### ISSUE TITLE
bug(radio): label isn't captured by macOS Hover Text feature

### ISSUE THREAD

**Issue by jelbourn** [angular/components#22950]
Steps to reproduce:

* Enable macOS Hover Text: https://support.apple.com/en-us/HT210978
* Try using hover text on a `<mat-radio-button>` (https://material.angular.io/components/radio)

The expected behavior is that the radio label gets magnified when you hover on the radio toggle itself (but not the label, which is strange but matches the native radio behavior)

The problem here is that macOS triggers the hover text on the native <input type="radio"> element, which we make very tiny with .cdk-visually-hidden. MDC radio doesn't have this problem, so it seems like the ideal solution would be reusing MDC's styles here without adding .cdk-visually-hidden at all.

It would be _nice_ to fix this for the non-MDC radio as well, but not strictly necessary.

**Comment by Splaktar** [comment]
The hover text feature works with our `mdc-radio`, but not `mat-radio`. So this seems to be "Fixed in MDC", but I will take a quick look at fixing `mat-radio` as well.

As another data point, this isn't a problem with AngularJS Material's radio buttons either: https://material.angularjs.org/latest/demo/radioButton

**Comment by jelbourn** [comment]
This issue was reported on our MDC-based radio, and I was definitely able to reproduce it on it. Which demo did you test?

**Comment by Splaktar** [comment]
I tested the MDC-radio on the `dev-app` at `/mdc-radio`.

I can confirm that removing `cdk-visually-hidden` fixes this.

This also affects `mat-checkbox`.

**Comment by Splaktar** [comment]
With the Hover Text feature, it seems to work best if you hold down ⌘ and then move the mouse around to hover over things to read. If you just move the mouse over something, then press and hold ⌘ w/o moving the mouse at all, often nothing happens until you move. This is true with the native "Enable Hover Text" checkbox in Settings as well as in web apps.

**Comment by Splaktar** [comment]
Here are examples of Hover Text working with the MDC radio and MDC checkbox in the dev-app:

https://user-images.githubusercontent.com/3506071/121928610-4f622380-cd0e-11eb-93b2-90b50977daaa.mov


https://user-images.githubusercontent.com/3506071/121928623-52f5aa80-cd0e-11eb-9515-c636497b6f7a.mov

**Comment by jelbourn** [comment]
@Splaktar you were right, I was mistaken- this issue was reported on the pre-MDC radio.

**Comment by Splaktar** [comment]
Good to know! There's a PR to posted to fix the mat-radio and mat-checkbox. I'll mark this as fixed in MDC as well.

**Comment by amysorto** [comment]
I just retested this on the MDC based radio and checkbox components and the hover text is working
